src/GitVersionCore.Tests/ExecuteCoreTests.cs

```diff
@@ -127,6 +127,17 @@ public void WorkingDirectoryWithoutGit()
         });
     }
 
+    [Test]
+    public void DynamicRepositoriesShouldNotErrorWithFailedToFindGitDirectory()
+    {
+        var versionAndBranchFinder = new ExecuteCore(fileSystem);
+
+        RepositoryScope(versionAndBranchFinder, (fixture, vv) =>
+        {
+            versionAndBranchFinder.ExecuteGitVersion("https://github.com/GitTools/GitVersion.git", null, new Authentication(), "refs/head/master", false, "tempProjectPath", null);
```

Comment:
If you look at the other tests in this file, `fixture.RepositoryPath` is what's used. `Environment.SystemDirectory` is used in the one test above just because it's guaranteed to:

1. Exist.
2. Not contain a `.git` folder.